hack/tools/tools.go

```diff
@@ -1,3 +1,6 @@
+//go:build tools
+// +build tools
```

Comment:
I don't mind this for consistency, but hack/tools isn't vendored, so this isn't really driving any behavior change currently, right?

Reply:
It matters in my Workspaces branch (I forget exactly why, but if you feel strongly I can remove this part for now), and I am trying to peel things out of there.

Reply:
hmm... that's disconcerting... `go list -e -tags=tools all` is the only place I know of in our scripts that should care about this tag, and that script doesn't look at this file. If go is attaching actual behavior to the `tools` tag now, I kind of want to know about it.

Let's drop it from this PR for now...

Reply:
Dredging memory, it was not that "tools" was special, but something about `go list` not liking imports of "main" packages (despite this being the documented hack for tools).  Using a tag forced it to not be a "package"  (no compilable code).  I am happy to pull this out of this PR and revisit in context.

Reply:
ah, ok. I'm still curious what needs `go list` to pass on this package

> I am happy to pull this out of this PR and revisit in context.

sgtm

Reply:
> I'm still curious what needs go list to pass on this dir

Memory is vague and a lot has changed.  Maybe it was `find . -name go.mod` coupled with "tell me what your module name is" -> error.  If so, it will come back and we can deal then.